esi/lib/nodes.py

```diff
@@ -138,3 +140,86 @@ def network_list(connection, filter_node=None, filter_network=None):
             })
 
     return data
+
+
+def network_attach(connection, node, filter_network=None, filter_port=None, filter_trunk=None, mac_address=None):
```

Comment:
I'd suggest changing these arguments. They aren't really filters; they're the actual networks/ports you want to use. In fact, it may feel cleaner to have something like:

`def network_attach(connection, node, network_dict)`

And then essentially have the same logic below, except checking the keys of the dict for network/port/trunk/mac_address.